scripts/license/add_license_header.sh

```diff
@@ -51,28 +51,60 @@ extend_license_header() {
 # root directory and all sub-directories.
 # Parameters:
 #   file extension, e.g.: .go, .cpp,
-#   a license text as a string
+#   comment prefix, e.g.: //, #,
 add_license_to_files() {
     local file_extension="$1"
-    local license_header="$2"
+    local prefix="$2"
+    local license_header="$(extend_license_header "$prefix")"
 
     # Get a list of all files in the project directory
-    local all_files=($(find $root_dir -type f -name "*$file_extension"))
+    mapfile -t -d $'\0' all_files < <(find "$root_dir" -type f -name "*$file_extension" -print0)
```

Comment:
what is `mapfile`? Is it available in bash, or does it need to install something extra - would be better not to need to install.

Reply:
It is in bash since 4.4 version. It should be instead of the old style. See [this](https://github.com/koalaman/shellcheck/wiki/SC2207)

Reply:
Switched it back because of macOS compatibility